torch/utils/cpp_extension.py

```diff
@@ -2422,3 +2481,518 @@ def _is_cuda_file(path: str) -> bool:
+        return library_dirs
+
+    def get_include_dirs(self):
+        include_dirs = []
+        include_dirs += [f"{x}" for x in self.get_sycl_include_dir()]
+        include_dirs += [f"{x}" for x in self.get_onemkl_include_dir()]
+        include_dirs += [f"{x}" for x in self.get_onednn_include_dir()]
+        include_dirs += [f"{x}" for x in self.get_default_include_dir()]
+        return include_dirs
+
+    def get_onemkl_libraries(self):
+        MKLROOT = self.__onemkl_root
+        return [
+            f"{MKLROOT}/lib/intel64/libmkl_sycl.a",
+            f"{MKLROOT}/lib/intel64/libmkl_intel_ilp64.a",
+            f"{MKLROOT}/lib/intel64/libmkl_sequential.a",
+            f"{MKLROOT}/lib/intel64/libmkl_core.a",
+        ]
+
+
+def _get_one_api_help():
+    oneAPI = _one_api_help()
+    return oneAPI
+
+def XPUExtension(name, sources, *args, **kwargs):
+    r"""
+    Creates a :class:`setuptools.Extension` for XPU/C++.
+    Convenience method that creates a :class:`setuptools.Extension` with the
+    bare minimum (but often sufficient) arguments to build a XPU/C++
+    extension.
+    All arguments are forwarded to the :class:`setuptools.Extension`
+    constructor.
+    Example:
+        >>> from torch.utils import XPUBuildExtension, XPUExtension
+        >>> setup(
+                name='xpu_extension',
+                ext_modules=[
+                    XPUExtension(
+                            name='xpu_extension',
+                            sources=['extension.cpp', 'extension_kernel.cpp'],
+                            extra_compile_args={'cxx': ['-g', '-std=c++20', '-fPIC']})
+                ],
+                cmdclass={
+                    'build_ext': XPUBuildExtension
+                })
+    """
+    library_dirs = kwargs.get("library_dirs", [])
+    library_dirs += library_paths()
+    kwargs["library_dirs"] = library_dirs
+
+    libraries = kwargs.get("libraries", [])
+    libraries.append("c10")
+    libraries.append("torch")
+    libraries.append("torch_cpu")
+    libraries.append("torch_python")
+
+    # Append oneDNN link parameters.
+    libraries.append("dnnl")
+    kwargs["libraries"] = libraries
+
+    include_dirs = kwargs.get("include_dirs", [])
+    include_dirs += include_paths(backend="xpu")
+    kwargs["include_dirs"] = include_dirs
+
+    kwargs["language"] = "c++"
+
+    extra_compile_args = kwargs.get("extra_compile_args", {})
+    extra_link_args = kwargs.get("extra_link_args", [])
+    # add oneapi link parameters
+    extra_link_args = _prepare_ldflags(extra_link_args, with_cuda=False, verbose=False, is_standalone=False)
+    extra_compile_args = _prepare_sycl_compile_flags(extra_compile_args)
+
+    # todo: add XPU parameter support.
+    kwargs["extra_link_args"] = extra_link_args
+    kwargs["extra_compile_args"] = extra_compile_args
+
+    return setuptools.Extension(name, sources, *args, **kwargs)
+
+class XPUBuildExtension(build_ext):
```

Comment:
This class does not make sense to me. I believe generic `BuildExtension` class should be extended instead.